internal/edge_cluster/edge_node_subresource.go

```diff
@@ -0,0 +1,137 @@
+// Copyright 2024 Broadcom. All Rights Reserved.
+// SPDX-License-Identifier: MPL-2.0
+
+package edge_cluster
+
+import (
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func EdgeNodeSchema() *schema.Resource {
+	return &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The name of the edge node",
+			},
+			"cluster_id": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The id of the compute cluster",
+			},
+			"admin_password": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The administrator password for the edge node",
+			},
+			"audit_password": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The audit password for the edge node",
+			},
+			"root_password": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The root user password for the edge node",
+			},
+			"tep1_ip": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The IP address (CIDR) of the first tunnel endpoint",
+			},
+			"tep2_ip": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The IP address (CIDR) of the second tunnel endpoint",
+			},
+			"tep_gateway": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The gateway for the tunnel endpoints",
+			},
+			"tep_vlan": {
+				Type:        schema.TypeInt,
+				Required:    true,
+				Description: "The VLAN ID for the tunnel endpoint",
+			},
+			"inter_rack_cluster": {
+				Type:        schema.TypeBool,
+				Required:    true,
+				Description: "Whether or not this is an inter-rack cluster. True for L2 non-uniform and L3, false for L2 uniform",
+			},
+			"management_ip": {
```

Comment:
Some ValidateFunc

Reply:
I added the ipv4 validation func form the util